src/frontend/src/components/EditUser/index.tsx

```diff
@@ -90,21 +190,18 @@ class EditUser extends React.Component<Props, State> {
                 className="w-100 pa3 br2 b--black-20"
                 required={true}
                 aria-describedby={
-                  this.state.invalidCredentials
-                    ? 'no_match_msg'
-                    : this.state.missingPassword
+                  this.state.missingPassword
                     ? 'input_msg'
                     : undefined
                 }
                 aria-invalid={
-                  this.state.invalidCredentials || this.state.missingPassword
+                  this.state.missingPassword
                     ? true
                     : false
                 }
                 onChange={this.handleChange}
               />
             </div>
-
             <div className="mb4">
```

Comment:
The change-role section shouldn't render if a non-admin user is updating their own account details.

You could use a similar thing as with the "Admin" button in the header, which is here:

https://github.com/codeforpdx/recordexpungPDX/blob/958feacf2d8b0e3abe1aedf99619f5aa35ab9313/src/frontend/src/components/Header/index.tsx#L34

It checks only a local variable which a user could hack to expose the "role" button, but the backend is protected to disallow a user to make themself admin, so it's fine.

Reply:
That would seem like a good idea. I went through and added that to the component. Thanks for the heads up.